Mathlib/Analysis/SpecialFunctions/MulExpNegMulSqIntegral.lean

```diff
@@ -0,0 +1,254 @@
+/-
+Copyright (c) 2025 Jakob Stiefel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Stiefel
+-/
+import Mathlib.Analysis.SpecialFunctions.MulExpNegMulSq
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+import Mathlib.MeasureTheory.Integral.BoundedContinuousFunction
+import Mathlib.MeasureTheory.Integral.DominatedConvergence
+import Mathlib.MeasureTheory.Measure.RegularityCompacts
+import Mathlib.Topology.ContinuousMap.StoneWeierstrass
+
+/-!
+# Properties of the integral of `mulExpNegMulSq`
+The mapping `mulExpNegMulSq` can be used to transform a function `g : E → ℝ` into a bounded
+function `mulExpNegMulSq ε ∘ g : E → ℝ = fun x => g x * Real.exp (-ε * g x * g x)`. This file
+contains results on the integral of `mulExpNegMulSq g ε` with respect to a finite measure `P`.
+## Lemmas
+- `tendsto_integral_mulExpNegMulSq_comp`: By the dominated convergence theorem and
+  `mulExpNegMulSq_abs_le_norm`, the integral of `mulExpNegMulSq ε ∘ g` with respect to a
+  finite measure `P` converges to the integral of `g`, as `ε → 0`;
+- `tendsto_integral_one_plus_inv_mul`: The integral of `mulExpNegMulSq ε ∘ g` with
+  respect to a finite measure `P` can be approximated by the integral of the sequence approximating
+  the exponential function, `fun x => (g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n) x`. This allows to
+  transfer properties of a subalgebra of functions containing `g` to the function
+  `mulExpNegMulSq ε ∘ g`, see e.g. `integral_mulExpNegMulSq_comp_eq`.
+## Main Result
+`dist_integral_mulExpNegMulSq_comp_le`: For a subalgebra of functions `A`, if for any `g ∈ A` the
+integral with respect to two finite measures `P, P'` coincide, then the difference of the integrals
+of `mulExpNegMulSq ε ∘ g` with respect to `P, P'` is bounded by `6 * sqrt ε`.
+This is a key ingredient in the proof of theorem `ext_of_forall_mem_subalgebra_integral_eq`,
+where it is shown that a subalgebra of functions that separates points separates finite measures.
+-/
+
+open MeasureTheory Real NNReal ENNReal BoundedContinuousFunction Filter
+
+open scoped Topology
+
+variable {E: Type*} [TopologicalSpace E] [MeasurableSpace E] [BorelSpace E]
+    (P : Measure E) [IsFiniteMeasure P]
+
+theorem integrable_mulExpNegMulSq_comp (f : C(E, ℝ)) {ε : ℝ} (hε : ε > 0) :
+    Integrable (mulExpNegMulSq ε ∘ f) P := by
```

Comment:
It would be slightly better to replace the compositions everywhere with a more applied form, as those are the ones that appear naturally in goals:
```suggestion
    Integrable (fun x ↦ mulExpNegMulSq ε (f x)) P := by
```
Also you sometimes use `=>` and sometimes `↦`. Please pick one for the whole file. My preference goes to `↦` but the style guide is not strict about it.